### PR TITLE
CCIP-4116 restructure tests

### PR DESCRIPTION
**Pull request by 0xsuryansh**
Test restructuring

### CODE REVIEW

File: contracts/src/v0.8/ccip/test/rmn/armProxy/ArmPorxy.setARM.t.sol [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.24;
+
+import {ARMProxy} from "../../../rmn/ARMProxy.sol";
+
+import {ARMProxyStandaloneTestSetup} from "./ARMProxyStandaloneTestSetup.t.sol";
+
+contract ARMProxySetARM is ARMProxyStandaloneTestSetup {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/rmn/armProxy/ArmPorxy.setARM.t.sol]: missing underscore between contract name and function being called, and should be lowercase `setARM`

File: contracts/src/v0.8/ccip/test/rmn/armProxy/ARMProxyStandaloneTestSetup.t.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.24;
+
+import {ARMProxy} from "../../../rmn/ARMProxy.sol";
+import {Test} from "forge-std/Test.sol";
+
+contract ARMProxyStandaloneTestSetup is Test {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/rmn/armProxy/ARMProxyStandaloneTestSetup.t.sol]: Can probably rename this to just ARMProxySetup to be in like with all other setups

File: contracts/src/v0.8/ccip/test/applications/PingPong/PingPong.plumbing.t.sol [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    assertEq(s_pingPong.getCounterpartChainSelector(), chainSelector);
+  }
+
+  function test_Fuzz_CounterPartAddress_Success(
+    address counterpartAddress
+  ) public {
+    s_pingPong.setCounterpartAddress(counterpartAddress);
+
+    assertEq(s_pingPong.getCounterpartAddress(), counterpartAddress);
```
Comment (jhweintraub) [inline contracts/src/v0.8/ccip/test/applications/PingPong/PingPong.plumbing.t.sol]: should be a separate file for each of these functions `getCounterPartAddress()`, `getCounterpartChainSelector`, and `isPaused`

File: contracts/src/v0.8/ccip/test/capability/CCIPHome/CCIPHome.applyChainConfigUpdates.t.sol [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      CAPABILITIES_REGISTRY,
+      abi.encodeWithSelector(INodeInfoProvider.getNodesByP2PIds.selector, chainReaders),
+      abi.encodeWithSelector(INodeInfoProvider.NodeDoesNotExist.selector, chainReaders[0])
+    );
+
+    vm.expectRevert(abi.encodeWithSelector(INodeInfoProvider.NodeDoesNotExist.selector, chainReaders[0]));
+    s_ccipHome.applyChainConfigUpdates(new uint64[](0), adds);
+  }
+
+  function test__applyChainConfigUpdates_FChainNotPositive_Reverts() public {
```
Comment (jhweintraub) [inline contracts/src/v0.8/ccip/test/capability/CCIPHome/CCIPHome.applyChainConfigUpdates.t.sol:186]: nit: double underscore

File: contracts/src/v0.8/ccip/test/capability/CCIPHome/CCIPHome.constructor.t.sol [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import {CCIPHomeTestSetup} from "./CCIPHomeTestSetup.t.sol";
+
+contract CCIPHome_constructor is CCIPHomeTestSetup {
+  function test_constructor_success() public {
+    CCIPHome ccipHome = new CCIPHome(CAPABILITIES_REGISTRY);
+
+    assertEq(address(ccipHome.getCapabilityRegistry()), CAPABILITIES_REGISTRY);
+  }
+
+  function test_supportsInterface_success() public view {
```
Comment (jhweintraub) [inline contracts/src/v0.8/ccip/test/capability/CCIPHome/CCIPHome.constructor.t.sol]: move to separate file for supportsInterface

File: contracts/src/v0.8/ccip/test/capability/CCIPHome/CCIPHome.constructor.t.sol [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    assertTrue(s_ccipHome.supportsInterface(type(IERC165).interfaceId));
+    assertTrue(s_ccipHome.supportsInterface(type(ICapabilityConfiguration).interfaceId));
+  }
+
+  function test_getCapabilityConfiguration_success() public view {
```
Comment (jhweintraub) [inline contracts/src/v0.8/ccip/test/capability/CCIPHome/CCIPHome.constructor.t.sol]: move out of file to getCapabilityConfiguration test file

**Review by jhweintraub — CHANGES_REQUESTED** [review]

File: contracts/src/v0.8/ccip/test/applications/PingPong/PingPong.plumbing.t.sol (hunk 3 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/test/applications/PingPong/PingPong.plumbing.t.sol]: https://github.com/smartcontractkit/chainlink/pull/15165/commits/ca63592985c489d791bb56212fc2850410c50715

File: contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter/FeeQuoter.applyDestChainConfigUpdates.t.sol [hunk 7]
```diff
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity 0.8.24;
 
-import {FeeQuoter} from "../../FeeQuoter.sol";
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter/FeeQuoter.applyDestChainConfigUpdates.t.sol]: FeeQuoter is a top level contract so you don't need a feeQuoter/FeeQuoter folder, just a FeeQuoter one

Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter/FeeQuoter.applyDestChainConfigUpdates.t.sol]: https://github.com/smartcontractkit/chainlink/pull/15165/commits/b039d1fbb0866525e68fc31785d48bdb06ef05c0

**Review by RensR — APPROVED** [review]

**Review by jhweintraub — APPROVED** [review]

**Pull request merged**